### PR TITLE
TH-1068: Adds 2 extra hobby types as category links to courses search

### PR DESCRIPTION
**Pull request by melniiv**
## Additional notes :spiral_notepad:

### CODE REVIEW

File: src/domain/landingPage/LandingPage.tsx [hunk 1]
```diff
@@ -45,9 +49,31 @@ const LandingPage: React.FC = () => {
...
+      courseCategories.concat(
+        hobbyTypes.filter(
+          (hobbyType) =>
+            hobbyType.value === COURSE_HOBBY_TYPES.CAMPS ||
+            hobbyType.value === COURSE_HOBBY_TYPES.CLUBS
+        ) || []
+      ),
+      'text'
+    );
+  };
```
[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/landingPage/LandingPage.tsx:67]: Would be little bit easier to add or delete included hobbyTypes in the future

File: src/domain/landingPage/landingPageSearchSection/LandingPageSearchSection.tsx [hunk 2]
```diff
@@ -66,10 +69,16 @@ const Search: React.FC<SearchProps> = ({
...
   };
 
-  const handleCategoryClick = (category: CategoryOption) => {
+  const handleCategoryClick = (category: CategoryExtendedOption) => {
+    //categories are extended to have hobby types, which has a different search query parameter.
+    const categoryParam = Object.values(COURSE_HOBBY_TYPES).includes(
+      category.value as COURSE_HOBBY_TYPES
+    )
+      ? 'hobbyTypes'
+      : 'categories';
```
Comment (Sartonon) [inline src/domain/landingPage/landingPageSearchSection/LandingPageSearchSection.tsx:78]: Maybe we should consider creating some constants for these query param strings.. If we happen to change the naming we easily forget to update them here.

Comment (Sartonon) [reply src/domain/landingPage/landingPageSearchSection/LandingPageSearchSection.tsx:78]: I'm thinking something like this:

_eventSearch/types.ts_
```

export const HOBBY_TYPES_PARAM = 'hobbyTypes';

export interface Filters {
  categories: string[];
  [HOBBY_TYPES_PARAM]?: string[];
  dateTypes: string[];
  divisions: string[];
  end: Date | null;
  isFree?: boolean;
  keyword?: string[];
  keywordNot?: string[];
  onlyChildrenEvents?: boolean;
  onlyEveningEvents?: boolean;
  alsoOngoingCourses?: boolean;
  places: string[];
  publisher?: string | null;
  start: Date | null;
  text: string[];
  audienceMinAgeGt?: string;
  audienceMaxAgeLt?: string;
}
```

Comment (Sartonon) [reply src/domain/landingPage/landingPageSearchSection/LandingPageSearchSection.tsx:78]: Of course all of them could be constants!

Comment (Sartonon) [reply src/domain/landingPage/landingPageSearchSection/LandingPageSearchSection.tsx:78]: Good example in linkedcomponents-ui by @jorilindell 

https://github.com/City-of-Helsinki/linkedcomponents-ui/blob/develop/src/domain/eventSearch/types.ts
https://github.com/City-of-Helsinki/linkedcomponents-ui/blob/develop/src/domain/eventSearch/constants.ts

**Review by Sartonon — CHANGES_REQUESTED** [review]
Looking pretty good, some changes should be done. Some improvement ideas too added.

File: src/domain/landingPage/LandingPage.tsx [hunk 3]
```diff
@@ -1,3 +1,4 @@
+import { orderBy } from 'lodash';
```
[the remark this answers is not in the document]
Comment (melniiv) [reply src/domain/landingPage/LandingPage.tsx]: fixed all.

File: src/domain/landingPage/LandingPage.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (melniiv) [reply src/domain/landingPage/LandingPage.tsx]: thanks. fixed.

File: src/domain/landingPage/landingPageSearchSection/LandingPageSearchSection.tsx (hunk 2 above)
Comment (melniiv) [reply src/domain/landingPage/landingPageSearchSection/LandingPageSearchSection.tsx]: fixed

**Review by Sartonon — APPROVED** [review]

**Pull request merged**